src/wp-includes/class-wp-term.php

```diff
@@ -107,11 +107,9 @@ final class WP_Term {
 	 * @global wpdb $wpdb WordPress database abstraction object.
 	 *
 	 * @param int    $term_id  Term ID.
-	 * @param string $taxonomy Optional. Limit matched terms to those matching `$taxonomy`. Only used for
-	 *                         disambiguating potentially shared terms.
-	 * @return WP_Term|WP_Error|false Term object, if found. WP_Error if `$term_id` is shared between taxonomies and
-	 *                                there's insufficient data to distinguish which term is intended.
-	 *                                False for other failures.
+	 * @param string $taxonomy Optional. Limit matched terms to those matching `$taxonomy`.
```

Comment:
Maybe clarify:
```suggestion
	 * @param string $taxonomy Optional. No longer used and silently deprecated.
```

Reply:
The behaviour of this parameter actually still applies, there's a test that covers it. It prevents a term in a different category being returned.